core/comm/client.go

```diff
@@ -213,3 +213,37 @@ func (client *GRPCClient) NewConnection(address string, tlsOptions ...TLSOption)
 	}
 	return conn, nil
 }
+
+type DynamicClientCredentials struct {
+	TLSConfig  *tls.Config
+	TLSOptions []TLSOption
+}
+
+func (dtc *DynamicClientCredentials) latestConfig() *tls.Config {
+	tlsConfigCopy := dtc.TLSConfig.Clone()
```

Comment:
Why Clone() ?

Reply:
Just for safety, to be extra cautious. 
This is a **tls.Config** so I make no assumptions on whether it can hold internal state or not by an existing TLS connection, and this has negligible computation overhead since it is just copying pointers anyway.

Reply:
Also note that we apply the mutator functions later on in this function, so I wanted the functions to be applied on a fresh shallow copy (which is deep enough of a copy, given the TLS config struct)